text/0000-hierarchic-anonymous-life-time.md

```diff
@@ -0,0 +1,151 @@
+- Feature Name: Hierarchic anonymous life-time
+- Start Date: 2020-06-24
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+New use of anonymous life-time `'_` that implicitly added to current structure.
+
+# Motivation
+
+Motivation is to simplify iterative development and improving refactoring of the code
+
+Sometimes during refactoring such code:
+```rust
+struct City {
+    name: String,
+    population: u32,
+}
+
+struct State {
+    city: Vec<City>,
+}
+
+struct Country {
+   state: Vec<State>,
+}
+```
+
+developer decides to make an inner item a reference:
+```rust
+struct City<'a> {
+    name: &'a str,
+    population: u32,
+}
+
+struct State<'a> {
+    city: Vec<City<'a>>,
+}
+
+struct Country<'a> {
+   state: Vec<State<'a>>,
+}
+```
+Everywhere in composition hierarchy I need to write 'a ... most of the times it is just boilerplate code ...
+
+What if instead of writing manually we will specify reference fields with anonymous life-time:
+```rust
+struct City& {
+    name: &str,
+    population: u32,
+}
+
+struct State& {
+    cities: Vec<City&>,
+}
+
+struct Country& {
+   state: Vec<State&>,
+}
```

Comment:
This looks uglier than the original one. I don't quite like the trailing `&`, no taste IMHO.